specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json

```diff
@@ -0,0 +1,1234 @@
+        },
+        "ip": {
+          "type": "string",
+          "description": "The IP exposed to the public internet."
+        },
+        "dnsNameLabel": {
+          "type": "string",
+          "description": "The Dns name label for the IP."
+        },
+        "fqdn": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The FQDN for the IP."
+        }
+      },
+      "required": [
+        "ports",
+        "type"
+      ]
+    },
+    "Port": {
+      "description": "The port exposed on the container group.",
+      "type": "object",
+      "properties": {
+        "protocol": {
+          "type": "string",
+          "description": "The protocol associated with the port.",
+          "enum": [
+            "TCP",
+            "UDP"
+          ],
+          "x-ms-enum": {
+            "name": "ContainerGroupNetworkProtocol",
+            "modelAsString": true
+          }
+        },
+        "port": {
+          "type": "integer",
+          "description": "The port number.",
+          "format": "int32"
+        }
+      },
+      "required": [
+        "port"
+      ]
+    },
+    "ContainerPort": {
+      "description": "The port exposed on the container instance.",
+      "type": "object",
+      "properties": {
+        "protocol": {
+          "type": "string",
+          "description": "The protocol associated with the port.",
+          "enum": [
+            "TCP",
+            "UDP"
+          ],
+          "x-ms-enum": {
+            "name": "ContainerNetworkProtocol",
+            "modelAsString": true
+          }
+        },
+        "port": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The port number exposed within the container group."
+        }
+      },
+      "required": [
+        "port"
+      ]
+    },
+    "EnvironmentVariable": {
+      "description": "The environment variable to set within the container instance.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the environment variable."
+        },
+        "value": {
+          "type": "string",
+          "description": "The value of the environment variable."
+        }
+      },
+      "required": [
+        "name",
+        "value"
+      ]
+    },
+    "OperationListResult":{
+      "description": "The operation list response that contains all operations for Azure Container Instance service.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Operation"
+          },
+          "description": "The list of operations."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URI to fetch the next page of operations."
+        }
+      }
+    },
+    "Operation": {
+      "description": "An operation for Azure Container Instance service.",
```

Comment:
same comment - this is used only as server return value. please mark all the properties as read-only.

Reply:
Done